src/components/Table.tsx

```diff
@@ -286,9 +286,9 @@ export default class Table<T> extends React.Component<
 	}
 
 	setRowSelection = (selectedRows: T[]): void => {
-		const rowKey = this.props.rowKey;
+		const { rowKey, data } = this.props;
 
-		if (!rowKey) {
+		if (!rowKey || !data) {
```

Comment:
How about moving this a bit lower, so that we can always do `setRowSelection([])`?
I'm thinking about the case that the table at some point had data and some of the rows were selected and at some point the property get's null. Even in that case the user should be able to try to clear the selections w/o issues.

Reply:
Good idea, I'll add a test case for this as well